db/db_impl/db_impl_open.cc

```diff
@@ -1263,14 +1263,32 @@ Status DBImpl::RecoverLogFiles(const std::vector<uint64_t>& wal_numbers,
       TEST_SYNC_POINT_CALLBACK(
           "DBImpl::RecoverLogFiles:BeforeUpdateProtectionInfo:batch",
           batch_to_use);
+
+      status = WriteBatchInternal::UpdateProtectionInfo(batch_to_use,
+                                                        /*bytes_per_key=*/8);
+      TEST_SYNC_POINT_CALLBACK(
+          "DBImpl::RecoverLogFiles:UpdateProtectionInfo::status", &status);
+      if (!status.ok()) {
+        if (status.IsCorruption()) {
+          reporter.Corruption(record.size(), status);
+          continue;
+        } else {
+          // Fail DB open for non-corruption failure.
+          return status;
+        }
+      }
```

Comment:
I stared at this for a while without understanding it. Maybe some comment will help: "`UpdateProtectionInfo()` examines the contents of the `WriteBatch` to calculate KV checksums. Any corruptions in the `WriteBatch` will be surfaced during this processing. Corruptions here indicate the `WriteBatch` we read was corrupted, so we follow the usual convention for reporting a `Corruption` in the input data. For cases where KV checksum detects a corruption introduced by the recovery process, see `VerifyChecksum()` below"

Reply:
I don't know if this is too complicated, but you might want to `VerifyChecksum()` even after detecting a corruption here to check whether the write batch became ill-formed by recovery corruption (and in that case do not use `reporter.Corruption()`).